lib/config.js

```diff
@@ -22,6 +22,12 @@ var autoconfig = function(overrides) {
       "sync_frequency" : SYNC_FREQUENCY,
       "storage_strategy": "dom",
       "do_console_log": false
+    },
+    syncFields: {
+      result: ['id', 'stepResults'],
```

Comment:
It's not mongoose specific or even a data-layer problem. I don't think we should be restricting the fields returned from the data layer to satisfy the synchronisation hashing functionality.

I had considered putting this logic inside the mongoose module, however it would be trying to solve the issues with sync in the data layer instead of the application.

It is up to the application developer to decide what fields are considered as part of synchronisation of client data.